src/backlight/portfolio/strategy.py

```diff
@@ -87,14 +87,14 @@ def _calculate_principals_lot_sizes(
         trade_currency = trade.currency_unit
 
         ratio = get_forex_ratio(trade.index[0], mkt, trade_currency, currency_unit)
-        #         if currency_unit == Currency[symbol[:3]]:
-        #             ratio2 = pd.Series(data=np.ones(trade.index.size), index=trade.index)
-        #         else:
-        #             ratio2 = get_forex_ratios(mkt, currency_unit,
-        #                     Currency[symbol[:3]]).loc[trade.index]
+        if currency_unit == Currency[symbol[:3]]:
```

Comment:
Do not use symbol[:3]. 

When symbol is `eurjpy` and currency_unit is `usd`,
ratio2 is exchange rate from `eur -> usd`.

You can calculate it by `eur -> jpy` x `jpy -> usd`.
Here, `rate` is `jpy -> usd` and `eur -> jpy` is `mkt.mid[trade.index[0]]`(= current price.).

1unit of eurjpy is `1eur`.
`1eur` -> `100jpy` -> `1usd`

Reply:
So, the lot size should be

```
lot_sizes[symbol] = principal * (current_price * rate) / (nb_trades * max_amount)
```

or 

```
lot_sizes[symbol] = principal / (current_price * rate) / (nb_trades * max_amount)
```